File: ruby/exercises/nth-prime/nth_prime.rb
# NthPrime
class NthPrime
  class << self
    def nth(num)
      fail ArgumentError if num < 1
      primes(num).last
    end

    def primes(num)
      found = []
      curr = 1
      while found.size < num
        found << curr if prime?(curr)
        curr += 1
      end
      found
    end

    def prime?(num)
      return false if num == 1
      return true if num == 2
      (2..Math.sqrt(num).ceil).each do |i|
        return false if num % i == 0
      end
      true
    end
  end
end

File: go/nth-prime/nth_prime.go
package prime

func Nth(n int) (int, bool) {
	if n <= 0 {
		return 0, false
	}
	primes := getPrimesUpTo(n)
	return primes[n-1], true
}

func getPrimesUpTo(n int) (primes []int) {
	candidate := 2
	for len(primes) != n {
		if isPrime(candidate) {
			primes = append(primes, candidate)
		}
		candidate += 1
	}
	return primes
}

func isPrime(x int) bool {
	for i := 2; i < x; i++ {
		if x%i == 0 {
			return false
		}
	}
	return true
}
